src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -1728,6 +1728,11 @@ private async Task ReconnectAsync(int timeout)
                         if (attempt == retryCount - 1)
                         {
                             SqlClientEventSource.Log.TryTraceEvent("<sc.SqlConnection.ReconnectAsync|INFO> Original ClientConnectionID {0} - give up reconnection", _originalConnectionId);
+                            if (e.Class >= TdsEnums.FATAL_ERROR_CLASS)
+                            {
+                                SqlClientEventSource.Log.TryTraceEvent("<sc.SqlConnection.ReconnectAsync|INFO> Original ClientConnectionID {0} - Fatal Error occured. Error Class: {1}", _originalConnectionId, e.Class);
+                                InnerConnection.CloseConnection(InnerConnection.Owner, ConnectionFactory);
```

Comment:
```suggestion
                                // Error Class: 20-25, usually terminates the database connection
                                InnerConnection.CloseConnection(InnerConnection.Owner, ConnectionFactory);
```
NIT: keeping it consistent with the comment from netcore